resources/opendal/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+use async_trait::async_trait;
+use opendal::{Operator, Scheme};
+use serde::{Deserialize, Serialize};
+use shuttle_service::{
+    error::{CustomError, Error as ShuttleError},
+    resource::Type,
+    Factory, IntoResource, ResourceBuilder, Secret,
+};
+use std::collections::{BTreeMap, HashMap};
+use std::str::FromStr;
+
+#[derive(Serialize)]
+pub struct Opendal {
+    scheme: String,
+}
+
+impl Default for Opendal {
+    fn default() -> Self {
+        Self {
+            scheme: "memory".to_string(),
+        }
+    }
+}
+
+impl Opendal {
+    pub fn scheme(mut self, scheme: &str) -> Self {
+        self.scheme = scheme.to_string();
+        self
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct OpendalOutput {
+    scheme: String,
+    cfg: BTreeMap<String, Secret<String>>,
+}
+
+pub struct Error(opendal::Error);
+
+impl From<Error> for shuttle_service::Error {
+    fn from(error: Error) -> Self {
+        let msg = format!("Failed to build opendal resource: {:?}", error.0);
+        ShuttleError::Custom(CustomError::msg(msg))
+    }
+}
+
+#[async_trait]
+impl ResourceBuilder for Opendal {
+    const TYPE: Type = Type::Custom;
+    type Config = Self;
```

Comment:
For this type of plugin it is better to use `()` as the config type (see some of the other plugins), since there is nothing that the shuttle servers should know about this resource.

Reply:
Fixed.